dist/tools/buildsystem_sanity_check/save_all_dependencies_resolution_variables.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2019 Freie Universität Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+# Script to save all the dependencies resolution variables for both normal and
+# in `info-boards-supported` resolution.
+#
+# The script takes a long time
+#
+# @author: Gaëtan Harter <gaetan.harter@fu-berlin.de>
+
+: "${RIOTBASE:="$(cd "$(dirname "$0")/../../../" || exit; pwd)"}"
+
+usage() {
+    echo "Usage: $0 <output_directory>"
+}
+
+
+applications() {
+    make --no-print-directory -C "${RIOTBASE}" info-applications
+}
+
+
+boards() {
```

Comment:
As this is implemented it will test against all boards, in #12092 this was limited by setting `ALLBOARDS`, I feel like it would be better if `boards` and `applications` where script optional arguments.

Reply:
I do not like duplicating the handling that is done in the build system.

Also, it should normally be executed on everything. It was only to save review time that I limited the boards, but I had to spend an hour justifying that only these boards were affected at the beginning of the testing procedure of #12092 
Also verify there was no side-effect from limiting the boards as there were before https://github.com/RIOT-OS/RIOT/pull/12041 and https://github.com/RIOT-OS/RIOT/pull/11478.

I would rather simplify it when the results are reliable enough.

If you want one board/application, the target can be run alone too. This one is the 'save_all' script.

For limiting applications, `makefiles/app_dirs.inc.mk` currently does not support it, and the implementation of `--applications` in `dist/tools/compile_and_test_for_board` had to compensate for it.

Reply:
Ok